public/app/plugins/datasource/graphite/migrations.ts

```diff
@@ -0,0 +1,33 @@
+import { GraphiteQuery } from './types';
+
+type LegacyAnnotation = {
+  target?: string;
+  tags?: string;
+};
+
+// this becomes the target in the migrated annotations
+const migrateLegacyAnnotation = (json: LegacyAnnotation): GraphiteQuery => {
+  return {
+    queryType: 'events',
+    eventsQuery: {
+      fromAnnotations: true,
+      target: json.target,
+      tags: json.tags,
+    },
+  } as GraphiteQuery;
```

Comment:
Instead of using `as GraphiteQuery`, can we return `"textEditor": true` if in case if the annotation is based on target string/query.

Reply:
Yes, I think this makes sense. We can not use ```as GraphiteQuery``` and add "textEditor": true if the annotation is based on a target string query. The GraphiteQuery was originally one field, 'target', so maybe we can make a new type for the annotations?